src/components/shared/LessonSide/LevelSelector.tsx

```diff
@@ -1,5 +1,42 @@
-function LevelSelector(): JSX.Element {
-  return <div>Level Selector</div>;
+import '../../../styles/LessonSide.scss';
+
+interface LevelSelectorProps {
+  numLevels: number;
+  currentLevel: number;
+}
+
+function LevelSelector({
+  numLevels,
+  currentLevel,
+}: LevelSelectorProps): JSX.Element {
+  return (
+    <div className="selector-box">
+      <div className="level-selector">
+        {Array(numLevels * 2 - 1)
+          .fill(0)
+          .map((_, i) => {
+            const level = i / 2 + 1;
+            return i % 2 == 1 ? (
+              <div
+                className={
+                  'level-connector' +
+                  (level < currentLevel ? ' reached-level-connector' : '')
+                }
+              ></div>
+            ) : (
+              <div
+                className={
+                  'level-button' +
+                  (level == currentLevel ? ' current-level' : '')
```

Comment:
`===` does exact equals without type conversion. See [here](https://codeahoy.com/javascript/2019/10/12/==-vs-===-in-javascript/#:~:text=The%20difference%20between%20%3D%3D%20and%20%3D%3D%3D%20is%20that%3A,same%20type%20before%20performing%20comparison.&text=%3D%3D%3D%20does%20not%20do%20any,the%20two%20variables%20being%20compared.) for details. Generally, defer to `===` if you don't know which to use
```suggestion
                  (level === currentLevel ? ' current-level' : '')
```